### PR TITLE
Add Integration samples to the repository

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose
Added the following integration samples

- Github issue summary to email
- NewsAPI headlines to email

### CODE REVIEW

File: github-issue-summary-to-email/main.bal [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import ballerina/http;
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Code is not properly formatted, use ballerina formatter.

File: github-issue-summary-to-email/main.bal [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import ballerina/log;
+import ballerinax/github;
+import wso2/choreo.sendemail as email;
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Why do we use choreo send mail? Can't we use `ballerina/email` module?

File: github-issue-summary-to-email/main.bal [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+configurable http:BearerTokenConfig gitHubTokenConfig = ?;
+configurable string repositoryName = ?;
+configurable string repositoryOwner = ?;
+
+// Email configuration parameters 
+configurable string recipientAddress = ?;
+
+public function main() returns error? {
+    string assigneeSummary = "";
+    github:Client githubClient = check new ({
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]:
```suggestion
    github:Client github = check new ({
```

Comment (anupama-pathirage) [reply github-issue-summary-to-email/main.bal]: This is the convention we use for clients

File: github-issue-summary-to-email/main.bal [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        auth: {
+            token: gitHubTokenConfig.token
+        }
+    });
+    stream<github:User,github:Error?> collaborators = check githubClient->getCollaborators(repositoryOwner, repositoryName);
+    log:printInfo("Started compiling the report");
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Unnecessary log line

File: github-issue-summary-to-email/main.bal [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import ballerina/http;
+import ballerina/log;
+import ballerinax/github;
+import wso2/choreo.sendemail as email;
+
+// Github client configuration parameters
+configurable http:BearerTokenConfig gitHubTokenConfig = ?;
+configurable string repositoryName = ?;
+configurable string repositoryOwner = ?;
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Isn't it better to use `orgName` and `repoName` instead of `repositoryOwner` and `repositoryName`?

File: github-issue-summary-to-email/main.bal [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    github:Client githubClient = check new ({
+        auth: {
+            token: gitHubTokenConfig.token
+        }
+    });
+    stream<github:User,github:Error?> collaborators = check githubClient->getCollaborators(repositoryOwner, repositoryName);
+    log:printInfo("Started compiling the report");
+    check collaborators.forEach(function (github:User user ){
+        string query = string `repo:${repositoryOwner}/${repositoryName} is:issue assignee:${user.login}`;
+        github:SearchResult|github:Error issuesForAssignee = githubClient-> search(query, github:SEARCH_TYPE_ISSUE, 1);
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Why only fetching 1?

File: github-issue-summary-to-email/main.bal [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        if issuesForAssignee is github:SearchResult {
+            string userName = user?.name ?: "Unknown Name";
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]:
```suggestion
            string userName = user?.name ?: "Unknown";
```

File: github-issue-summary-to-email/main.bal [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    }
+    int totalClosedIssueCount = closedIssues is github:SearchResult? closedIssues.issueCount :0;
+
+    string issueSummary = string `ISSUE SUMMARY REPORT${"\n\n"}Repository Name: ${repositoryName}
+        ${"\n"}Total Issues Open: ${totalOpenIssueCount} ${"\n"}Total Issues Closed: ${totalClosedIssueCount}
+        ${"\n\n"}Issue Count by Assignee: ${"\n"}${assigneeSummary} ${"\n"}`;
+    email:Client emailClient = check new ();
+    string sendEmailResponse = check emailClient->sendEmail(recipientAddress, "Git Issue Summary", issueSummary);
+    log:printInfo("Email sent successfully \n " + sendEmailResponse);
+}
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal:69]: Since the example is bit too wordy add a comment line before each major step. like `//Get collaborator list`, `//Get open issues`, `//Get closed issues`, `//Send email` etc

File: newsapi_headlines-to-email/Readme.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+[<ORG_NAME>.newsapi_headlines_to_email.apiKeyConfig]
+apiKey = "<NEWSAPI_API_KEY>"
+```
+
+## Testing
+Run the Ballerina project by executing `bal run` from the root.
+
+Once successfully executed, BBC's top headlines are fetched and sent as an email to the recipient.
+
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/Readme.md:33]: Extra new line.

File: newsapi_headlines-to-email/main.bal [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import ballerina/log;
+import ballerinax/newsapi;
+import wso2/choreo.sendemail;
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: Why using choreo?

File: newsapi_headlines-to-email/main.bal [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+//News API configuration
+configurable newsapi:ApiKeysConfig apiKeyConfig = ?;
+configurable string emailAddress = ?;
+
+public function main() returns error? {
+    string mailBody = "";
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: Define this inline with the change in https://github.com/ballerina-guides/integration-samples/pull/35/files#r1310480457

File: newsapi_headlines-to-email/main.bal [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import ballerina/log;
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal:2]: Code is not formatted properly.

File: newsapi_headlines-to-email/main.bal [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+//News API configuration
+configurable newsapi:ApiKeysConfig apiKeyConfig = ?;
+configurable string emailAddress = ?;
+
+public function main() returns error? {
+    string mailBody = "";
+    newsapi:Client newsClient = check new (apiKeyConfig, {}, "https://newsapi.org/v2");
+    newsapi:WSNewsTopHeadlineResponse topHeadlines = check newsClient->listTopHeadlines(sources="bbc-news", page=1);
+    log:printInfo(topHeadlines.toString());
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: No need to print these.

File: newsapi_headlines-to-email/main.bal [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    newsapi:WSNewsArticle[]? articles = topHeadlines?.articles;
+    if articles is newsapi:WSNewsArticle[] && articles.length() != 0 {
+        mailBody = "BBC top news are,\n";
+        foreach var article in articles {
+            string? title = article?.title;
+            if title is string {
+                mailBody = mailBody + string `\t* ${title}\n`;
+                log:printInfo(mailBody);
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: unnecessary log.

File: newsapi_headlines-to-email/main.bal [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+public function main() returns error? {
+    string mailBody = "";
+    newsapi:Client newsClient = check new (apiKeyConfig, {}, "https://newsapi.org/v2");
+    newsapi:WSNewsTopHeadlineResponse topHeadlines = check newsClient->listTopHeadlines(sources="bbc-news", page=1);
+    log:printInfo(topHeadlines.toString());
+    newsapi:WSNewsArticle[]? articles = topHeadlines?.articles;
+    if articles is newsapi:WSNewsArticle[] && articles.length() != 0 {
+        mailBody = "BBC top news are,\n";
+        foreach var article in articles {
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: Avoid using var.

File: newsapi_headlines-to-email/main.bal [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+            string? title = article?.title;
+            if title is string {
+                mailBody = mailBody + string `\t* ${title}\n`;
+                log:printInfo(mailBody);
+            }
+        }
+        sendemail:Client sendemailEndpoint = check new ();
+        string sendEmailResponse = check sendemailEndpoint->sendEmail(emailAddress, "BBC Headlines", mailBody);
+        log:printInfo("Email sent successfully!" + sendEmailResponse);
+    } else {
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: Avoid else by using early returns.

File: newsapi_headlines-to-email/main.bal [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+configurable newsapi:ApiKeysConfig apiKeyConfig = ?;
+configurable string emailAddress = ?;
+
+public function main() returns error? {
+    string mailBody = "";
+    newsapi:Client newsClient = check new (apiKeyConfig, {}, "https://newsapi.org/v2");
+    newsapi:WSNewsTopHeadlineResponse topHeadlines = check newsClient->listTopHeadlines(sources="bbc-news", page=1);
+    log:printInfo(topHeadlines.toString());
+    newsapi:WSNewsArticle[]? articles = topHeadlines?.articles;
+    if articles is newsapi:WSNewsArticle[] && articles.length() != 0 {
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal]: better to have as `if articles is () || articles.length() == 0 {`

File: newsapi_headlines-to-email/main.bal [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+                log:printInfo(mailBody);
+            }
+        }
+        sendemail:Client sendemailEndpoint = check new ();
+        string sendEmailResponse = check sendemailEndpoint->sendEmail(emailAddress, "BBC Headlines", mailBody);
+        log:printInfo("Email sent successfully!" + sendEmailResponse);
+    } else {
+        log:printInfo("No news found");
+    }
+}
```
Comment (anupama-pathirage) [inline newsapi_headlines-to-email/main.bal:41]: With all above suggestion, my version of sample is as follows.

```ballerina
public function main() returns error? {
    newsapi:Client newsClient = check new (apiKeyConfig, {}, "https://newsapi.org/v2");
    newsapi:WSNewsTopHeadlineResponse topHeadlines = check newsClient->listTopHeadlines(sources = "bbc-news", page = 1);
    newsapi:WSNewsArticle[]? articles = topHeadlines?.articles;
    if articles is () || articles.length() == 0 {
        log:printInfo("No news found");
        return;
    } 
    string mailBody = "BBC top news are,\n";
    foreach newsapi:WSNewsArticle article in articles {
        string? title = article?.title;
        if title is string {
            mailBody = mailBody + string `\t* ${title}\n`;
        }
    }
    sendemail:Client sendemailEndpoint = check new ();
    string sendEmailResponse = check sendemailEndpoint->sendEmail(emailAddress, "BBC Headlines", mailBody);
    log:printInfo("Email sent successfully!" + sendEmailResponse);
}

```

File: github-issue-summary-to-email/main.bal (hunk 3 above)
Comment (chathurace) [reply github-issue-summary-to-email/main.bal]: @anupama-pathirage it's better if we document all these in a best-practices page. I couldn't find a doc with import ordering conversion as well.

File: newsapi_headlines-to-email/main.bal (hunk 10 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Choreo sendmail was used for the ease of testing these examples for users. Compared to the ballerina email connector where a host need to be setup separately, choreo.sendmail can be used to send the email without any additional effort.

File: github-issue-summary-to-email/main.bal (hunk 2 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Choreo sendmail was used for the ease of testing these examples for users. Compared to the ballerina email connector where a host need to be setup separately, choreo.sendmail can be used to send the email without any additional effort.

File: github-issue-summary-to-email/main.bal (hunk 6 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: As this query here always returns a result containing a summary, it only need to fetch 1 result.

File: github-issue-summary-to-email/main.bal (hunk 1 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081).

File: github-issue-summary-to-email/main.bal (hunk 2 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 3 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 4 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 5 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 7 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 8 above)
Comment (sahanHe) [reply github-issue-summary-to-email/main.bal:69]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/Readme.md (hunk 9 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/Readme.md:33]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 10 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 11 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 12 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal:2]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 13 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 14 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 15 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 16 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 17 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal]: Updated in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: newsapi_headlines-to-email/main.bal (hunk 18 above)
Comment (sahanHe) [reply newsapi_headlines-to-email/main.bal:41]: Updated with some additional changes in [5131705](https://github.com/ballerina-guides/integration-samples/pull/35/commits/51317055927a041c645786b20bcb6af3864f1081)

File: github-issue-summary-to-email/main.bal (hunk 3 above)
Comment (anupama-pathirage) [reply github-issue-summary-to-email/main.bal]: > @anupama-pathirage it's better if we document all these in a best-practices page. I couldn't find a doc with import ordering conversion as well.

@chathurace We have those in https://learn-ballerina.github.io/ currently - an unofficial list 

Naming conversions : https://learn-ballerina.github.io/best_practices/naming_conventions.html#clients

Import ordering : https://learn-ballerina.github.io/best_practices/format_the_code.html#order-of-the-imports-in-a-bal-file

File: github-issue-summary-to-email/main.bal [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+configurable string smtpPassword = ?;
+configurable string smtpUsername = ?;
+configurable string smtpHost = ?;
+
+// Email configuration parameters 
+configurable string recipientAddress = ?;
+configurable string fromAddress = ?;
+
+public function main() returns error? {
+    string assigneeSummary = "";
```
Comment (anupama-pathirage) [inline github-issue-summary-to-email/main.bal]: Move the variable declaration to the closest usage.

Comment (sahanHe) [reply github-issue-summary-to-email/main.bal]: Updated in [462bb79](https://github.com/ballerina-guides/integration-samples/pull/35/commits/462bb7951567e84f94fb0d009f1ff15abfc2d92b)

**Review by anupama-pathirage — APPROVED** [review]
